module/VuFind/src/VuFind/Content/PageLocator.php

```diff
@@ -84,35 +84,102 @@ public function __construct($themeInfo, $language, $defaultLanguage)
         $this->defaultLanguage  = $defaultLanguage;
     }
 
+    /**
+     * Generate a template from a file search pattern. Examples:
+     * - %pathPrefix%/%pageName%{_%language%} => content/help_en
+     * - %pathPrefix%/%language%/%pageName% => HelpTranslations/en/search
+     *
+     * @param string $pathPrefix Subdirectory where the template should be located
+     * @param string $pageName   Page name
+     * @param string $pattern    Filesystem pattern
+     * @param string $language   Language
+     *
+     * @return string
+     */
+    protected function generateTemplateFromPattern(
+        $pathPrefix, $pageName, $pattern, $language=null
+    ) {
+        $template = $pattern;
+        $template = str_replace('%pathPrefix%', $pathPrefix, $template);
+        $template = str_replace('%pageName%', $pageName, $template);
+        $languagePatternExtended = '"\\{(.*)(%language%)(.*)\\}"';
+        if (isset($language)) {
+            $template = preg_replace(
+                $languagePatternExtended,
+                "\\1$language\\3",
+                $template
+            );
+        } else {
+            $template = preg_replace($languagePatternExtended, '', $template);
+        }
+        $template = str_replace('%language%', $language ?? '', $template);
+        $template = str_replace('//', '/', $template);
+        return $template;
+    }
+
+    /**
+     * Try to find a template using
+     * 1) Current language suffix
+     * 2) Default language suffix
+     * 3) No language suffix
+     *
+     * @param string $pathPrefix Subdirectory where the template should be located
+     * @param string $pageName   Template name
+     * @param string $pattern    Filesystem pattern
+     *
+     * @return array Array with template options (key equals matchType)
+     */
+    protected function getTemplateOptionsFromPattern(
+        $pathPrefix, $pageName, $pattern
+    ) {
+        $templates = ['language' => $this->generateTemplateFromPattern(
```

Comment:
I think this would be more readable as

```
$templates = [
    'language' => $this->generateTemplateFromPattern(
        $pathPrefix, $pageName, $pattern, $this->language
    )
];
```